src/Symfony/Component/DependencyInjection/Compiler/CheckDefinitionValidityPass.php

```diff
@@ -48,6 +48,14 @@ public function process(ContainerBuilder $container)
                 if ($definition->getFactory()) {
                     throw new RuntimeException(sprintf('Please add the class to service "%s" even if it is constructed by a factory since we might need to add method calls based on compile-time checks.', $id));
                 }
+                if (class_exists($id) || interface_exists($id)) {
```

Comment:
Why that? We want to check whether the interface can be loaded by the project's autoloader, not whether it has been loaded during compilatiom.